### ISSUE TITLE
Visual Studio ignores analyzer ProjectReferences when building System.Text.Encodings.Web

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/runtime#68763]
Visual Studio 17.2 Preview 5
SDK: 6.0.100

Repro:
```
git clone https://github.com/ViktorHofer/runtime && cd runtime
git checkout UpdateSlnsWithClosure
.\build.cmd libs
.\build.cmd -vs System.Text.Encodings.Web
```

Build src\System.Text.Encodings.Web inside VS and observe in the output window that the referenced "LibraryImportGenerator" and "Microsoft.Interop.SourceGeneration" projects aren't built along with System.Text.Encodings.Web even though a ProjectReference to them exist.

### Correct behavior
If I change [this line](https://github.com/dotnet/runtime/blob/9f2add03cae252e9e1e6af88a7d0260a6c722915/eng/generators.targets#L29) in generators.targets in which the ProjectReference to the two analyzer projects is defined to
```
                        and ('$(TargetFrameworkIdentifier)' == '.NETStandard' or '$(TargetFrameworkIdentifier)' == '.NETFramework' or ('$(TargetFrameworkIdentifier)' == '.NETCoreApp'))" />
```

the projects get built along with S.T.E.W, as expected. I inspected the design time builds and the projects are returned via the `ResolveProjectReferencesDesignTime2`:

![image](https://user-images.githubusercontent.com/7412651/166234551-27c299c6-427c-46d6-b3b8-f20b8a2f4cd4.png)

I experimented with avoiding the item function and just use `$(TargetFramework.StartsWith('net6.0'))` but that doesn't work either.

@drewnoakes, cc @ericstj

**Comment by ViktorHofer** [comment]
sorry for looping you in @drewnoakes but I believe this is an issue on our side. We have custom tooling that filters uncompatible target frameworks out and if I disable that tooling by setting `<UseTargetFrameworkPackage>false</UseTargetFrameworkPackage>` in the project file, I see the analyzer projects being built as well.

**Comment by ViktorHofer** [comment]
OK I found an easier way to reproduce the issue and I'm now certain that it's not our fault but probably the project system or VS.

When changing the order of tfms in https://github.com/dotnet/runtime/blob/9f2add03cae252e9e1e6af88a7d0260a6c722915/src/libraries/System.Text.Encodings.Web/src/System.Text.Encodings.Web.csproj#L3 so that `$(NetCoreAppMinimum)` is first, I see the analyzer projects being built. If `$(NetCoreAppCurrent)` is first, the analyzer projects aren't built.

**Comment by ericstj** [comment]
@ViktorHofer given what you know now, could you produce a simplified repro that just uses `TargetFramework` and `ProjectReference` items with conditions?  If we can prove it that simply then we can probably open a project system bug on it.

**Comment by ViktorHofer** [comment]
Yes, here's a minimal repro: [p2pvs.zip](https://github.com/dotnet/runtime/files/8604129/p2pvs.zip)
